crates/ruff_linter/src/docstrings/sections.rs

```diff
@@ -452,7 +428,19 @@ fn is_docstring_section(
     previous_section: Option<&SectionContextData>,
     previous_line: Option<&Line>,
     next_line: Option<&Line>,
+    definition: &Definition<'_>,
 ) -> bool {
+    // for function definitions, track the known argument names for more accurate section detection.
+    let known_args: FxHashSet<_> = definition
+        .as_function_def()
+        .map(|func| {
+            func.parameters
+                .iter()
+                .map(|param| param.name().as_str())
+                .collect()
+        })
+        .unwrap_or_default();
```

Comment:
Having this at the top here is sort of confusing, considering that it's only use is 70 lines further down. I suggest moving the logic closer to where it is. I'd also argue that computing a hash set isn't worth it, considering that functions tend to have only few parameters and we only perform one `contains` lookup at most (constructing a HashSet is `O(n)`, we can just do a linear search)

The alternative is to move the hash set calculation out of `is_docstring_section` and only run it once for the entire function.